cli/commands/project_config.go

```diff
@@ -80,22 +81,55 @@ func NewProjectConfigFromFile(filename string) *ProjectConfig {
 
 // Ensures our configuration data structure conforms to our ad hoc schema.
 // @TODO do this in a more formal way. See docker/libcompose for an example.
-func ValidateProjectConfig(config *ProjectConfig) error {
-	if len(config.Version) == 0 {
+func (c *ProjectConfig) ValidateConfigVersion() error {
+	if len(c.Version) == 0 {
 		return fmt.Errorf("No 'version' property detected.")
 	}
 
-	if config.Version != "1.0" {
-		return fmt.Errorf("Version '1.0' is the only supported value, found '%s'.", config.Version)
+	if c.Version != "1.0" {
+		return fmt.Errorf("Version '1.0' is the only supported value, found '%s'.", c.Version)
 	}
 
-	if config.Scripts != nil {
-		for id, script := range config.Scripts {
-			if len(script.Run) > 10 {
-				util.Logger().Warning.Printf("Project script '%s' has more than 10 run items (%d). You should create a shell script to contain those.", id, len(script.Run))
+	return nil
+}
+
+// Validate the config scripts against a set of rules/norms
+func (c *ProjectConfig) ValidateProjectScripts(subcommands []cli.Command) {
+	logger := util.Logger()
+
+	if c.Scripts != nil {
+		for id, script := range c.Scripts {
+
+			// Check for an empty script
+			if script == nil {
+				logger.Error.Fatalf("Project script '%s' has no configuration", id)
+			}
+
+			// Check for scripts with conflicting aliases with existing subcommands or subcommand aliases
+			for _, subcommand := range subcommands {
+				if id == subcommand.Name {
+					logger.Error.Fatalf("Project script name '%s' conflicts with command name '%s'. Please choose a different script name", id, subcommand.Name)
```

Comment:
The downside of "throwing" the error here is that if someone is in a working project directory, and runs `rig start` a broken outrigger.yml file will cause rig to bail. Or am I wrong?

That's what I was trying to get at with the idea of surfacing messages and an error state for the caller to decide what to do. I suppose we want `rig project` to choke on this, but we don't really care if a given outrigger.yml file is broken for other commands.